comp/forwarder/defaultforwarder/forwarder_health.go

```diff
@@ -142,22 +157,40 @@ func (fh *forwarderHealth) healthCheckLoop() {
 	}
 }
 
-// computeDomainsURL populates a map containing API Endpoints per API keys that belongs to the forwarderHealth struct
-func (fh *forwarderHealth) computeDomainsURL() {
+func (fh *forwarderHealth) updateAPIKey(newKey string) {
+	for _, dr := range fh.domainResolvers {
+		dr.UpdateAPIKey(fh.mainAPIKey, newKey)
+	}
```

Comment:
Could we have the domain forwarder not rely on the `forwarderHealth` to receive configuration notification ? I think the less coupled we have those moving part the better.
It looks like we can update the config notification to notify with the old and new value and each part could be independent, no ?

Reply:
Yes, makes sense. I updated the code as described.